### PR TITLE
feat: Issue credential - changed message structures according to v2

### PR DESCRIPTION
**Pull request by soluchok**
In issue credential v2, protocol messages were changed. See this [RFC](https://github.com/hyperledger/aries-rfcs/blob/master/features/0453-issue-credential-v2/README.md)

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/issuecredential/models.go [hunk 1]
```diff
@@ -20,23 +20,18 @@ type ProposeCredential struct {
...
-	// SchemaVersion is an optional filter to request credential based on a schema version.
-	// This is useful to allow a more user-friendly experience of requesting a credential by schema name and version.
-	SchemaVersion string `json:"schema_version,omitempty"`
-	// CredDefID is an optional filter to request credential based on a particular Credential Definition.
-	// This might be helpful when requesting a commercial driver's license instead of
-	// an ordinary driver's license, for example.
-	CredDefID string `json:"cred_def_id,omitempty"`
-	// IssuerDid is an optional filter to request a credential issued by the owner of a particular DID.
-	IssuerDid string `json:"issuer_did,omitempty"`
+	// Formats contains an entry for each filter~attach array entry, providing the the value of the attachment @id
```
Comment (troyronda) [inline pkg/didcomm/protocol/issuecredential/models.go:23]: where are you updating the protocol version number?

Comment (soluchok) [reply pkg/didcomm/protocol/issuecredential/models.go:23]: I missed that

File: pkg/didcomm/protocol/issuecredential/models.go [hunk 2]
```diff
@@ -20,23 +20,18 @@ type ProposeCredential struct {
...
+	// and the verifiable credential format and version of the attachment.
+	Formats []Format `json:"formats,omitempty"`
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/issuecredential/models.go:25]: just curious - don't we have any bdd tests to show usage of these attributes ?

Comment (soluchok) [reply pkg/didcomm/protocol/issuecredential/models.go:25]: No, the usage depends on the specific implementation.

**Pull request merged**